app/domain/authentication/authn_azure/validate_application_identity.rb

```diff
@@ -0,0 +1,84 @@
+require 'forwardable'
+require 'command_class'
+
+module Authentication
+  module AuthnAzure
+
+    Err = Errors::Authentication::AuthnAzure
+
+    ValidateApplicationIdentity = CommandClass.new(
+        dependencies: {
+            role_class:                ::Role,
+            resource_class:             Resource,
+            application_identity_class: ApplicationIdentity
+        },
+        inputs:       %i(account, service_id, username, xms_mirid, oid)
+    ) do
+
+      def call
+        parse_token_field
+        populate_identity_values
+        # compare xms_mirid with what is defined in annotations
+        validate_role_annotations
+        # check integrity of annotations
+        validate_application_annotations
+      end
+
+      private
+
+      # traverse xms_mirid field for later value comparisons
+      def parse_token_field
+        field_split = xms_mirid.split('/')
+        @xms_mirid_hash = Hash[field_split.each_slice(2).to_a]
```

Comment:
I purposely separated out this logic so that 1 function does not do a handful tasks but _also_ felt it was awkward. Adding it back in as you suggested